src/platform-camera.cpp

```diff
@@ -72,7 +72,15 @@ platform_camera::platform_camera( std::shared_ptr< const device_info > const & d
     std::vector< std::shared_ptr< platform::uvc_device > > devs;
     auto backend = get_backend();
     for( auto & info : uvc_infos )
-        devs.push_back( backend->create_uvc_device( info ) );
+    {
+        auto & dev = environment::get_instance().get_uvc_device( info.unique_id );
+        if( ! dev )
+        {
+            dev = get_backend()->create_uvc_device( info );
```

Comment:
This actually does a set...
I.e., the next line means nothing.
But this is also dangerous if multiple threads are in this same loop.